ig/input/fsh/nbs-msh-segment.fsh

```diff
@@ -3,15 +3,35 @@ Id: segment-msh-logical-model
 Title: "MSH Segment"
 Description: "The MSH Segment"
 * segmentType 1..1 SU string "MSH"
-* delimeters 1..1 SU string "MSH-1 The delimeters for the message"
+* fieldSeparator 1..1 SU string "MSH-1 The field separator for the message"
+* encodingCharacters 1..1 SU string "MSH-2 Encoding Characters"
 * sendingApplication 1..1 SU string "MSH-3 The sending application"
+* sendingApplicationNamespaceId 1..1 SU string "MSH-3.1 Sending Application Namespace ID"
+* sendingApplicationUniversalId 1..1 SU string "MSH-3.2 Sending Application Universal ID"
+* sendingApplicationUniversalIdType 1..1 SU string "MSH-3.3 Sending Application Universal ID Type"
 * sendingFacility 1..1 SU string "MSH-4 The sending facility"
+* sendingFacilityNamespaceId 1..1 SU string "MSH-4.1 The sending facility namespace ID"
+* sendingFacilityUniversalId 1..1 SU string "MSH-4.2 The sending facility universal ID"
+* sendingFacilityUniversalIdType 1..1 SU string "MSH-4.3 The sending application universal ID type"
 * receivingApplication 1..1 SU string "MSH-5 The receiving application"
+* receivingApplicationNamespaceId 1..1 SU string "MSH-5.1 The receiving application namespace ID"
+* receivingApplicationUniversalId 1..1 SU string "MSH-5.2 The receiving application universal ID type"
+* receivingApplicationUniversalIdType 1..1 SU string "MSH-5.3 The receiving application universal ID type"
 * receivingFacility 1..1 SU string "MSH-6 The receiving facility"
+* receivingFacilityNamespaceId 1..1 SU string "MSH-6.1 Receiving Facility Namespace ID"
+* receivingFacilityUniversalId 1..1 SU string "MSH-6.2 Receiving Facility Universal ID"
+* receivingFacilityUniversalIdType 1..1 SU string "MSH-6.3 Receiving Facility Universal ID Type"
 * messageDateTime 1..1 SU string "MSH-7 The date and time the message was created"
-* security 0..1 SU string "indicates if security features should be followed"
 * messageType 1..1 SU string "MSH-9 The type of message"
+* messageTypeMessageCode 1..1 SU string "MSH-9.1 Message Code"
+* messageTypeTriggerEvent 1..1 SU string "MSH-9.2 Trigger Event"
+* messageTypeMessageStructure 1..1 SU string "MSH-9.3 Message Structure"
 * messageControlId 1..1 SU string "MSH-10 a control id for the message"
 * processingId 1..1 SU string "MSH-11 the processing id"
 * versionId 1..1 SU string "MSH-12 the HL7 version of this message"
-// there are other optional fields we might want to define
\ No newline at end of file
+* acceptAcknowledgmentType 1..1 SU string "MSH-15 Accept Acknowledgment Type"
+* applicationAcknowledgementType 1..1 SU string "MSH-16 Application Acknowledgement Type"
+* messageProfileIdentifier 1..2 SU string "MSH-21 Message Profile Identifier"
```

Comment:
Data type for MSH-21: `EI`

Reply:
There's a data type called [`Identifier`](https://build.fhir.org/datatypes.html#Identifier) . Not sure if this is the same, but it may line up for the IG; Thoughts?

Reply:
I think that will work, great job!